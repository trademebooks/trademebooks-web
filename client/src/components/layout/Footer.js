import React from 'react'
import { MDBCol, MDBContainer, MDBRow, MDBFooter } from 'mdbreact'
import { useLocation } from 'react-router-dom'

import './Footer.scss'

const Footer = () => {
  const location = useLocation()

  // If the current URL is not in this list, then show the footer in that URL
  const showFooter = ![
    // black list of URLs
    'chat' // e.g on the /chat/** URL, the footer will NOT show
  ].includes(location.pathname.split('/')[1])

  return (
    <>
      {showFooter ? (
        <MDBFooter className="font-small pt-3 pb-1 teal">
          <MDBContainer className="text-center text-md-center">
            <MDBRow>
              <MDBCol md="3">
                <a href="/">
                  <h5>TRADEMEBOOKS</h5>
                </a>
              </MDBCol>
              <MDBCol md="6">
                <ul className="footer-links">
                  <li>
                    <a href="/about">About</a>
                  </li>
                  <li>
                    <a href="/contact">Contact Us</a>
                  </li>
                  <li>
                    <a href="/add-book">Post a Book</a>
                  </li>
                  <li>
                    <a href="/terms">Terms</a>
                  </li>
                  <li>
                    <a href="/privacy">Privacy</a>
                  </li>
                </ul>
              </MDBCol>
              <MDBCol md="3">
                &copy; {new Date().getFullYear()}{' '}
                <a href="https://www.chosensolutions.ca">Chosen Solutions</a>.
              </MDBCol>
            </MDBRow>
          </MDBContainer>
        </MDBFooter>
      ) : (
        <></>
      )}
    </>
  )
}

export default Footer
