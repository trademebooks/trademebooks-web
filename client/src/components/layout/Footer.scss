footer {
  .footer-links > li {
    list-style-type: none;
    display: inline-block;
    margin-left: 1rem;
  }
}
