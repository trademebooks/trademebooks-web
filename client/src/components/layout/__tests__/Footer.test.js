import React from 'react'
import renderer from 'react-test-renderer'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import Footer from '../Footer'

describe('Component: Footer', () => {
  it('should match the snapshot', () => {
    const history = createMemoryHistory()
    const route = '/some-test-route'
    history.push(route)

    const snapshot = renderer
      .create(
        <Router history={history}>
          <Footer />
        </Router>
      )
      .toJSON()

    expect(snapshot).toMatchSnapshot()
  })
})
