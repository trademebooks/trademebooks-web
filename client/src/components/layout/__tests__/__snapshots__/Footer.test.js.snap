// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer should match the snapshot 1`] = `
<footer
  className="page-footer font-small pt-3 pb-1 teal"
  data-test="footer"
>
  <div
    className="container text-center text-md-center"
    data-test="container"
  >
    <div
      className="row"
      data-test="row"
    >
      <div
        className="col-md-3"
        data-test="col"
      >
        <a
          href="/"
        >
          <h5>
            TRADEMEBOOKS
          </h5>
        </a>
      </div>
      <div
        className="col-md-6"
        data-test="col"
      >
        <ul
          class="footer-links"
        >
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/contact"
            >
              Contact Us
            </a>
          </li>
          <li>
            <a
              href="/add-book"
            >
              Post a Book
            </a>
          </li>
          <li>
            <a
              href="/terms"
            >
              Terms
            </a>
          </li>
          <li>
            <a
              href="/privacy"
            >
              Privacy
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-md-3"
        data-test="col"
      >
        Â© 
        2021
         
        <a
          href="https://www.chosensolutions.ca"
        >
          Chosen Solutions
        </a>
        .
      </div>
    </div>
  </div>
</footer>
`;
